src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs

```diff
@@ -0,0 +1,97 @@
+﻿using AutoMapper;
+using Microsoft.Dynamics.CRM;
+using Spd.Resource.Repository.Alias;
+using Spd.Resource.Repository.Application;
+using Spd.Resource.Repository.BizLicApplication;
+using Spd.Resource.Repository.LicApp;
+using Spd.Resource.Repository.Org;
+using Spd.Utilities.Dynamics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Spd.Resource.Repository.ControllingMemberCrcApplication;
+public class ControllingMemberCrcRepository : IControllingMemberCrcRepository
+{
+    private readonly DynamicsContext _context;
+    private readonly IMapper _mapper;
+
+
+    public ControllingMemberCrcRepository(IDynamicsContextFactory ctx, IMapper mapper)
+    {
+        _context = ctx.CreateChangeOverwrite();
+        _mapper = mapper;
+    }
+
+    public async Task<ControllingMemberCrcApplicationCmdResp> CreateControllingMemberCrcApplicationAsync(CreateControllingMemberCrcAppCmd cmd, CancellationToken ct)
+    {
+        // get parrent business license application
+        spd_application? bizLicApplication = await _context.spd_applications
+                .Expand(a=>a.spd_businessapplication_spd_workerapplication)
+                .Expand(a => a.spd_ApplicantId_contact)
+                .Expand(a => a.spd_ApplicantId_account)
+                .Where(a => a.spd_applicationid == cmd.ParentBizLicApplicationId)
+                .SingleOrDefaultAsync(ct);
+
+        if (bizLicApplication == null)
+            throw new ArgumentException("Original business licence application was not found.");
+
+        // create controlling member application
+        spd_application app = _mapper.Map<spd_application>(cmd);
+        _context.AddTospd_applications(app);
+        // create contact
+        contact contact = _mapper.Map<contact>(cmd);
+        contact.contactid = Guid.NewGuid();
+        _context.AddTocontacts(contact);
+
+        await _context.SaveChangesAsync(ct);
+
+        var account = _context.accounts
+            .Where(a => a.accountid == bizLicApplication.spd_ApplicantId_account.accountid)
+            .Where(a => a.statecode == DynamicsConstants.StateCode_Active)
+            .FirstOrDefault();
+        if (account != null)
+        {
+            _context.SetLink(app, nameof(spd_application.spd_OrganizationId), account);
+        }
+
+        _context.SetLink(app, nameof(app.spd_ApplicantId_contact), contact);
+        await _context.SaveChangesAsync(ct);
```

Comment:
do not do saveChanges here

Reply:
removed (in dynamics, before we want to add or set a link, it does not matter that the object has saved and has a primary value?)

Reply:
yeah, as long as you already addToxxx, Please test if you remove saveChanges, it is still working normally